AfterClass-io/afterclass.io-v2
# feat(ui): filter section responsive mobile viewport

closes #202

## Changes

- make checkbox responsive
- make all filter section subcomponent responsive
- fix scrolling to top issue when filtering for prof/ course
- make mobile filter section show only 3 items by default, expandable on click

In this change, I've introduced a new hook `useBreakpoint` that is basically a css media query that runs on javascript in the browser. It's not recommended to use this hook unless absolutely required

## Testing the changes

1. go to https://afterclass-io-v2-git-202-afterclass.vercel.app/course/ACCT102
2. open mobile viewport
3. should see filter section collapse to top 3 items
4. click on "show more"
5. should see "show more" -> "show less", filter section should expand to show all
6. clik on any professor to filter for him/her
7. should not scroll to top, should see review section be filtered for the professor


## Preview / Screenshots




| <img width="200px"/>  | before | after |
|:-:|:-:|:-:|
|mobile<br>professor page| ![screenshot_20240824_172042](https://github.com/user-attachments/assets/86d01f84-e974-404d-b62d-6d71164af085)| ![screenshot_20240824_172109](https://github.com/user-attachments/assets/e8440fc9-0098-4939-aa71-3818995b6f14) |
|mobile<br>course page| ![screenshot_20240824_173629](https://github.com/user-attachments/assets/9cb860d3-c0ab-4567-8b4f-8eea976d0061) | ![screenshot_20240824_173637](https://github.com/user-attachments/assets/b06c714d-c056-453d-90b3-3a593baaad5d) |
|desktop<br>professor page|  ![screenshot_20240824_172445](https://github.com/user-attachments/assets/dfc496c1-9ad5-4505-b463-1be57f0995dd) | ![screenshot_20240824_172441](https://github.com/user-attachments/assets/aef9a3f1-ec9d-44fc-b6c4-e65d506e125f) | 
|desktop<br>course page| ![screenshot_20240824_173333](https://github.com/user-attachments/assets/8a50f7aa-c66f-4f91-8f54-6222dcfa06c2) | ![screenshot_20240824_173327](https://github.com/user-attachments/assets/6990eb18-845c-412f-b74a-80470148179e) |



## Checklist
